File: backend/libs/store/src/objects/search_github_project.rs
use sea_orm::{prelude::Uuid, FromQueryResult};

#[derive(Deserialize, Serialize, ToSchema, FromQueryResult)]
#[serde(rename_all = "camelCase")]
pub struct SearchGithubProject {
    pub id: Uuid,
    pub name: String,
    pub repositories: i64,
    pub languages_used: Vec<String>,
    pub issues: i64,
    pub stargazers_count: i64
}

File: frontend/src/types/searchGithubProject.ts
export interface SearchGithubProject {
    id: string;
    name: string;
    repositories: number;
    languagesUsed: string[];
    issues: number;
    stargazersCount: number;
}

export interface SearchGithubProjectParams {
    page?: number;
    perPage?: number;
    order?: string;
    orderBy?: string;
    languagesUsed?: string[];
}